### DISCUSSION TITLE
WingetUI 3.0.0 Beta 0

### DISCUSSION THREAD

**Discussion by marticliment** [Devolutions/UniGetUI#1842] [General]
![](https://img.shields.io/github/downloads/marticliment/wingetui/3.0.0-beta0/total?style=for-the-badge&color=red)

## Changes
 - This is the first beta release. All the missing features since 3.0-alpha2 have been ported. 
 - Added Package Tags
 - Improvements on the Settings UI.
 - Lots of bugfixes and minor details.

SHA256: `ee3e041b21b9b7b01a5545debdf2dc51520bb46d1a1bca6859139e9761b480d6`

<hr /><em>This discussion was created from the release <a href='https://github.com/marticliment/WingetUI/releases/tag/3.0.0-beta0'>WingetUI 3.0.0 Beta 0</a>.</em>

**Comment by rumplin** [comment]
I didn't use any of the WingetUI 3.x.x versions before, this beta is my first instal lof 3.x.x.
So after this gets a proper release will my beta version get the automatic update to the latest version when it get's released?

**Reply by marticliment** [reply]
In theory, yes

**Comment by iLL888** [comment]
Hello, well done.
Here is some additional feedback for Beta.
_** - Bugs or issues that disrupts usability._


Beta Version
- **Duplicate items can end up in queue, request ability to remove duplicates.
- **Searching for packages doesn't mention items already in queue. (only on active search) Can result in duplicate queue items.
- Request the  "Select All" option to have ability to "Select un-acquired" to avoid items already installed and in queue.
- **The Installed, and Error icons are hard to read. Request the failed icon to be a different icon color.
- Live output shows odd characters:  Eg, ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-'ā-' 70%
- Display how many items remain in queue, without having to scroll the queue down to see.
- Scrollbar in queue is too close to 'Cancel' button, using the scrollbar can risk pressing cancel.
- **The ability to remember which items have 'failed' on further searches during the app session.
- Toggle hide/show sidebar.

Requests from previous version carried forward.
- When software fails because the hash does not match, adding a option 'Skip integrity check'. Include warning if needed.
- Successfully installed queue bar auto disappears even when the mouse is still hovered over the item.
- The ability to change the order of the queue items by adding the ability to move item up/down, or move to top.
- **The ability to pause all queue items.
- **The ability to remember queue items after you exit WingetUI.
- Would be useful to have a button to easily open the Temp directory where the installation files are downloaded.
- The progress bar for search (blue horizontal bar) never stops moving even when the search has completed.
- Search Mode (sidebar) does not have a default option selected. (Once selected, how to deselect?)
- If you can add back the ability to increase the queue window size.

**Reply by marticliment** [reply]
Hello @iLL888,

I will take your suggestions into account

**Comment by itayking1990** [comment]
Thank you! The beta version working great!

Please consider adding a hamburger menu with a line indicator and animation when moving between sections in the hamburger menu.

This is WingetUI Beta:
![image](https://github.com/marticliment/WingetUI/assets/158079551/c6447043-ebd6-45a3-900c-1fbfdcde464e)

DevToy example for line indicator based on the computer setting accent color with live hamburger 

As you can see the line indicator color is based on the user accent color and is also different from light mode (line indicator accent color is darker) and dark mode (line indicator accent color is lighter).
Also if the window size shrinks the hamburger menu shows only icons and if there is enough space it shows icons and text anyway there is the hamburger menu button that can change this behavior on the fly.

![image](https://github.com/marticliment/WingetUI/assets/158079551/f2dcf619-9aa0-4395-96e6-67c3846add5a)

![image](https://github.com/marticliment/WingetUI/assets/158079551/97291c54-d39c-4d2e-9346-6be95b11afac)


![image](https://github.com/marticliment/WingetUI/assets/158079551/512420ba-a54d-43c5-b934-96482fa2bf8a)

**Reply by itayking1990** [reply]
It is a little offensive when you comment on everyone else except me I took time and effort to post this UI / UX improvement.

I do not say you have to implement them but a little comment that you consider it would be much appreciated 

And for the record, this side menu is the official one from Microsoft guidelines to fluent design

**Reply by marticliment** [reply]
I am sorry, either your post didn't pop up on my inbox or I just accidentally skipped it. I have been receiving lots of emails recently (50/70 emails per day) and it is difficult to answer them all.

I actually read your post actually tried implementing the accent color border bar, but it didn't look good, it just felt out of place, and decided to not implement it. 

Regarding the breadcumb bar, since there are only 7 entries, and they are used often, I decided for a ms-store like nav bar, rather than the standard navbar that hides labels, and since buttons are smaller it is clicked more difficultly. 

I am sorry for not answering back, I have had tons of work and I might just have accidentally  skipped yours.

**Reply by itayking1990** [reply]
Thank you very much for taking the time to answer me in detail

**Comment by CanePlayz** [comment]
Any idea why WingetUI doesn't run without a manual installation of the Windows App Runtime? The runtimes are pre-installed:

![image](https://github.com/marticliment/WingetUI/assets/81419612/b901a609-ec60-4bae-b93b-d6848aa1ffd4)

**Reply by marticliment** [reply]
The instaler should install the runtimes by default, how are you running WingetUI?

**Reply by CanePlayz** [reply]
Yeah, the installer does that. When I uninstall this instance that was installed by the installer, WingetUI doesn't run anymore despite the AppRuntime being installed as you can clearly see in the winget screenshot. This is the error message I get:

![image](https://github.com/marticliment/WingetUI/assets/81419612/019e63df-f932-451f-b834-fed66158e0e8)

The installed version (from the winget screenshot) also seems to be the required one.

**Reply by marticliment** [reply]
The Windows App Runtime includes various packages, so while a package listed as "app runtime" might be installed, not all the dependencies may be present after you remove one of them

**Reply by CanePlayz** [reply]
Yeah thanks, maybe the DDLM is not included in this default installed package, because a special DDLM entry shows up under "Installed apps" when you install the runtime manually.

**Comment by rumplin** [comment]
There is no notification in the Beta 0 that a new version is availalbe (Beta 1).

**Reply by marticliment** [reply]
The updater does not work across prerelease versions, it only works for stable releases. All beta releases will be updated to release when it is available
